engine/runtime/src/main/java/org/enso/interpreter/node/typecheck/AllOfTypesCheckNode.java

```diff
@@ -54,6 +70,17 @@ Object executeCheckOrConversion(VirtualFrame frame, Object value, ExpressionNode
       values[at] = result;
       at++;
     }
+    if (at != checks.length) {
+      // request for Number & Integer may yield two integers collision
+      // request for Number & Float may yield two floats collision
+      // request for Number & Integer & Float must yield one collision
```

Comment:
There are [tests in Conversion_Spec](https://github.com/enso-org/enso/blob/develop/test/Base_Tests/src/Semantic/Conversion_Spec.enso#L298) that combine these types. Thus such a combination must be supported (and it clashes with [assert checking each type is in multi value only once](https://github.com/enso-org/enso/actions/runs/12607741286/job/35139480421#step:7:1227)) - but it doesn't have to be fast. Thus `transferToInterpreter()`.